core/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -307,9 +339,28 @@ public boolean exists(Settings settings) {
      * instead.
      */
     public T get(Settings settings) {
+        return get(settings, true);
+    }
+
+    private T get(Settings settings, boolean validate) {
         String value = getRaw(settings);
         try {
-            return parser.apply(value);
+            T parsed = parser.apply(value);
+            if (validate) {
```

Comment:
We always have a validator as we [initialize](https://github.com/jasontedor/elasticsearch/blob/91f7f56b3fbdcc56802f66e90442d260f92c3b7b/core/src/main/java/org/elasticsearch/common/settings/Setting.java#L219) with a trivial validator in cases one is not provided when the `Setting` is defined.